adapta/storage/delta_lake/README.md

```diff
@@ -26,11 +27,22 @@ adls_path = AdlsGen2Path.from_hdfs_path('abfss://container@account.dfs.core.wind
 # get Iterable[pandas.DataFrame]
 batches = load(azure_client, adls_path, batch_size=1000)
 
-# create a filter and apply it
-filter = (pyarrow_field("my_column") == "some-value")  # case sensitive!
-filtered = load(azure_client, adls_path, row_filter=filter, columns=["my_column", "my_other_column"])
+# create generic filter expressions
+simple_filter = FilterField[str]("my_column") == "some-value"
+combined_filter = (FilterField[str]("my_column") == "some-value") & (FilterField[str]("other_column") == "another-value")
+combined_filter_with_collection = (FilterField[str]("my_column") == "something1") & (FilterField[str]("other_column").isin(['else', 'nonexistent']))
+complex_filter = (FilterField[str]("my_column") == "something1") | (FilterField[str]("other_column") == "else") & (FilterField[int]("another_column") == 123)
 
-# filtered is of type pandas.DataFrame
+# compile filters for PyArrow
+simple_expression_pyarrow = compile_expression(simple_filter, ArrowFilterExpression)
```

Comment:
Can we avoid doing the compilation manually? Would be nice if it could be invoked inside the load function based on a provided socket instead of needing to call either delta load or astra load based on the socket contents manually.

Reply:
> Can we avoid doing the compilation manually? Would be nice if it could be invoked inside the load function based on a provided socket instead of needing to call either delta load or astra load based on the socket contents manually.

I support this, we should simple change expression types for `load` and `filter_entities` to `Union...` and compile if needed